### ISSUE TITLE
Any support for deployment rollback or listing revisions?

### ISSUE THREAD

**Issue by cuiyuancao** [kubernetes-client/csharp#437]
I could not find any API method to perform deployment rollback operation and I could not find any deployment revision/history fields in the current V1Deployment class to apply a patch as an alternative way. Any suggested approach please? Thanks!

**Comment by tg123** [comment]
do you mean `resourceVersion` for history?

retrieve older version and patch === rollback

**Comment by tg123** [comment]
i think you are talking about `deployment.kubernetes.io/revision` 
it is in annotation

**Comment by cuiyuancao** [comment]
Thanks for pointing that out. I did not find it previously because I used to call "PatchNamespacedDeployment" and that did not generate an revision (of course). 

After I switched to "ReplacednamespacedDeployment", I am able to see deployment annotation like the following in V1ObjectMeta object in the V1Deployment object:
            "annotations": [
                {
                    "key": "deployment.kubernetes.io/revision",
                    "value": "3"
                },
                {
                    "key": "kubectl.kubernetes.io/last-applied-configuration",
                    "value": "{\"apiVersion\":\"apps/v1\",\"kind\":\"Deployment\",.................}}}\n"
                }
            ]

I am now able to call the following code to convert "last-applied-configuration" into V1Deployment object and replace the current deployment with it:  
            var lastAppliedConfigStr = prodDeployment.Metadata.Annotations.Where(m => m.Key == "kubectl.kubernetes.io/last-applied-configuration").First().Value;
            var lastAppliedDeployment = JsonConvert.DeserializeObject<V1Deployment>(lastAppliedConfigStr);
            var updatedDeployment = k8s.ReplaceNamespacedDeployment(lastAppliedDeployment, prodDeploymentName, aksNamespace);
 
It works well. Just wondering if there is any other easier way to do it like simply passing the revision value? My approach can only roll back to the last revision but cannot roll back to any history version unless I keep rolling back in a loop.  

If there is no other suggested way, I will close this issue later. Thank you!

**Comment by cuiyuancao** [comment]
Thanks for pointing that out. I did not find it previously because I used to call "PatchNamespacedDeployment" and that did not generate a revision.

After I switched to "ReplaceNamespacedDeployment", I am able to see deployment annotation like the following in V1ObjectMeta object in the V1Deployment object:
"annotations": [
{
"key": "deployment.kubernetes.io/revision",
"value": "3"
},
{
"key": "kubectl.kubernetes.io/last-applied-configuration",
"value": "{"apiVersion":"apps/v1","kind":"Deployment",.................}}}\n"
}
]

I am wondering if the current C# client API provides a way to rollback based on revision value? In my case I would like to call a method to rollback the deployment to version "2". In Python client, the following code is supported:

def rollback_deployment(api_instance, revision=0):
    rollback_to = client.AppsV1beta1RollbackConfig(revision)
    body = client.AppsV1beta1DeploymentRollback(
        name=DEPLOYMENT_NAME,
        rollback_to=rollback_to)
    api_response = api_instance.create_namespaced_deployment_rollback(
        name=DEPLOYMENT_NAME,
        namespace="default",
        body=body)

Thanks!
